source/group_eq_bot/driver/local/reader.py

```diff
@@ -0,0 +1,49 @@
+from dataclasses import dataclass
+from typing import Union
+from pathlib import Path
+from io import BufferedReader
+
+from driver.local.audio.base import BaseAudioReader
+from driver.local.text.base import BaseTextReader
+from driver.models.supported_extensions import  SupportedFilesExtensions
+
+
+@dataclass
+class Reader:
+    path: Path  # full path to a particular file to open
+
+    def __post_init__(self) -> None:
+        self.path = Path(self.path)
+
+    def check_file_existance(self) -> None:
+        """ Function, which checks whether file exist in given path and is a file """
+
+        if not self.path.exists():
+            raise FileNotFoundError(f'{self.path} path does not exist')
+        if not self.path.is_file():
+            raise FileNotFoundError(f'{self.path} is not a file')
+
+    def matches_supported_extensions(self, file_extension: str) -> None:
```

Comment:
1. D suggest to rename function as `def extension_supported(self, extension: str)`
2. I presume, it would be better function will return something if all is correct
3. Why function returns `None` in case extension is correct ?
4. Why function returns ` FileNotFoundError` error in case extension is INcorrect ?
5. function typing is not full (missing Error) 
To wrap:
```python
def extension_supported(self, extension: str) -> Union[bool, OSError]:
    return True if extension in SupportedFilesExtensions.list() else OSError(f'File extension is not supported by the reader. Supported ones are: {SupportedFilesExtensions.list()}')
```

Reply:
Renamed, added bool and errors to returns and to typings

One-liner does not work the same way, so I left the extended way